src/VPolygon.jl

```diff
@@ -241,27 +241,83 @@ If the direction has norm zero, the first vertex is returned.
 
 ### Algorithm
 
-This implementation performs a brute-force search, comparing the projection of
-each vector along the given direction.
-It runs in ``O(n)`` where ``n`` is the number of vertices.
-
-### Notes
-
-For arbitrary points without structure this is the best one can do.
-However, a more efficient approach can be used if the vertices of the polygon
-have been sorted in counter-clockwise fashion.
-In that case a binary search algorithm can be used that runs in ``O(\\log n)``.
-See issue [#40](https://github.com/JuliaReach/LazySets.jl/issues/40).
+This implementation uses a binary search algorithm when the polygon has more
+than ten vertices and a brute-force search when it has ten or less.
+For the brute-force search, it compares the projection of
+each vector along the given direction, it runs in ``O(n)`` where 
+``n`` is the number of vertices.
+For the binary search the algorithm runs in ``O(log n)``.
 """
-function σ(d::AbstractVector{N}, P::VPolygon{N}) where {N<:Real}
+function σ(d::AbstractVector{N}, P::VPolygon{N}) where {N <: Real}
     @assert !isempty(P.vertices) "the polygon has no vertices"
-    i_max = 1
+    if length(P.vertices) > 10
+        @inline _up(u, v) = dot(u, v) > 0
```

Comment:
I would prefer to pull out these inlined functions from `σ(d::AbstractVector{N}, P::VPolygon{N})`. Moreover, let's create wrap the code binary search code and the "brute force" code inside their own functions (compare to the design of `convex_hull`).

Reply:
I moved out the inlined functions, if you have a better place to put them, let me know!

Reply:
You can put them in `src/Arrays/vector_operations.jl` (you need to rebase with `master` first).